### ISSUE TITLE
Get Feature Sequence does not translate first codon as Met

### ISSUE THREAD

**Issue by erasche** [GMOD/Apollo#1544]
This mostly applies to non-table1 organisms, but sometimes we have L starts. They should be translated as a Met when in first position.

![utvalg_136](https://cloud.githubusercontent.com/assets/458683/24927369/4dab8158-1eee-11e7-9fa5-f61c8037d0cb.png)

**Comment by nathandunn** [comment]
fyi @erasche  This PR does not fix it.  @monicacecilia  is compiling our combined @monicacecilia / @deepakunni3  discussion.

**Comment by nathandunn** [comment]
@erasche  This might fix it, but I haven't really tested with it:

```
diff --git a/src/groovy/org/bbop/apollo/sequence/SequenceTranslationHandler.groovy b/src/groovy/org/bbop/apollo/sequence/SequenceTranslationHandler.groovy
index a28b976..acc1759 100644
--- a/src/groovy/org/bbop/apollo/sequence/SequenceTranslationHandler.groovy
+++ b/src/groovy/org/bbop/apollo/sequence/SequenceTranslationHandler.groovy
@@ -74,6 +74,10 @@ class SequenceTranslationHandler {
         for (int i = 0; i + 3 <= sequence.length(); i += 3) {
             String codon = sequence.substring(i, i + 3);
             String aminoAcid = translationTable.translateCodon(codon);
+            if(i==0 && translationTable.isStartCodon(codon)){
+                aminoAcid = "M"
+            }
+
             if (aminoAcid.equals(TranslationTable.STOP)) {
                 if (includeStop) {
                     buffer.append(aminoAcid);
diff --git a/src/groovy/org/bbop/apollo/sequence/TranslationTable.groovy b/src/groovy/org/bbop/apollo/sequence/TranslationTable.groovy
index 97fdc27..69236e1 100644
--- a/src/groovy/org/bbop/apollo/sequence/TranslationTable.groovy
+++ b/src/groovy/org/bbop/apollo/sequence/TranslationTable.groovy
@@ -86,6 +86,7 @@ class TranslationTable {
         return alternateTranslationTable;
     }
 
-
-
+    boolean isStartCodon(String s) {
+        return startCodons.contains(s)
+    }
 }
```

**Comment by monicacecilia** [comment]
Hello, 
@erasche, you are correct - at the moment the translation table service in Apollo does not force non-M starts to be __M__ (or __fM__ for that matter on table 11); they remain the amino acid that they code on the selected table. Though I am positive that the editing logic is still using the instructions for alternate initiation to find the longest ORF. 

Regardless, it is an error. Alternate start codons should still be translated as __M__ residues in the amino acid sequence; frankly, these being rather rare cases and not having had much active annotation for prokaryotes in recent past, the issue had not been raised. I am guilty of being quite arthropod-centric myself. 

Thanks for pointing this out. At the moment we will not be able to jump into modifying the results of the translated sequence to incorporate this change; all I can offer with our apologies is an invitation to take a stab at interpreting the special condition of translating marked codons into __Met__, when present in the first position, for the particular translation table being used. Otherwise, we may be able to address it again after the end of May. 

@nathandunn please remember to use these as reference:
https://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html/index.cgi?chapter=tgencodes or https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi

**Comment by monicacecilia** [comment]
errr... or, apparently, Nathan can take a stab right now... 😶

**Comment by erasche** [comment]
Tested this morning, that patch works perfectly

**Comment by erasche** [comment]
Opened a PR for you to save you the trouble: https://github.com/GMOD/Apollo/pull/1546

**Comment by monicacecilia** [comment]
Great to hear this worked. 
I'm re-opening to keep in on the horizon for re-testing before next release. This is not a 'pending' issue, but rather one that will require further testing.

**Comment by nathandunn** [comment]
Agreed.

However, all of the closed issues for 2.0.7 need to be tested before we release.   So closing it will be the most consistent.

**Comment by nathandunn** [comment]
Closing as appearing to work.  We will retest with the other closed issues when we release 2.0.7.
